### ISSUE TITLE
Strange behavior when running a binary file

### ISSUE THREAD

**Issue by pwd491** [flet-dev/flet#3055]
**Description**
I'm trying to localize my program. 
I am using the gettext module. When compiling the project, I add the `locales `directory with the `flat pack main.py --add-binary locales:locales` to use it after, but I encountered strange behavior. 
If I run the program using python, it works as expected, the locale is located in a folder in the project, but if we try to run a binary file that creates and extracts its files to a temporary folder at startup, then the gettext module cannot find the locale.

**For example:** 
The project tree looks like this:
```
D:\DEVELOPER\I10N
│   build.sh
│   flet.log
│   main.py
│
└───locales
    │   base.pot
    │
    └───ru_RU
        └───LC_MESSAGES
                base.mo
                base.po
```
**It works properly in this way:** `python ./main.py `
**Logs:** 
```
DEBUG:root:Path to file execute d:\Developer\i10n\main.py
DEBUG:root:Path to locales for translate: d:\Developer\i10n\locales
DEBUG:root:Is found locales? d:\Developer\i10n\locales\ru_RU\LC_MESSAGES\base.mo
DEBUG:root:Is dir exists? True
```

**I'm trying to compile an application and do this:**
`flet pack main.py --add-binary locales:locales`
**and than...**
try to open `main.exe`
**Logs:**
```
DEBUG:root:Path to file execute C:\Temp\_MEI85922\main.py
DEBUG:root:Path to locales for translate: C:\Temp\_MEI85922\locales
DEBUG:root:Is found locales? None
DEBUG:root:Is dir exists? True
```
I really don't understand why this behavior happens, it only happens on Windows and Mac OS, the binary file is executed correctly only on Linux.

**Code example to reproduce the issue:**
### I'll leave the archive with the project. Don't worry, it's as small as possible, created only to demonstrate behavior.
```python
import os
import gettext
import logging
import flet as ft

logging.basicConfig(filename="D:\\Developer\\i18n\\flet.log", level=logging.DEBUG)
logging.getLogger("flet_core").setLevel(logging.INFO)

async def main(page: ft.Page):
    path_to_locales = os.path.join(os.path.dirname(__file__), "locales")

    logging.debug("\n")
    logging.debug("Path to file execute %s", __file__)
    logging.debug("Path to locales for translate: %s", path_to_locales)
    logging.debug("Is found locales? %s", gettext.find("base", path_to_locales))
    logging.debug("Is dir exists? %s", os.path.isdir(path_to_locales))
    logging.debug("\n")

    translation = gettext.translation("base", path_to_locales, fallback=True)
    _ = translation.gettext

    page.title = _("Translate title of Application")
    page.update()

ft.app(main)
```

**Describe the results you received:**
Undetected language files when executing a binary file.

**Describe the results you expected:**
Localization of the application when using a binary file.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Flet version (`pip show flet`)**:
```
0.22.0
```

**Give your `requirements.txt` file (don't `pip freeze`, instead give direct packages):**

```
flet==0.22.0
pyinstaller==6.6.0
```

**Operating system**:
Windows, macOS.

**Additional environment details:**

**Comment by FeodorFitsner** [comment]
So, the issue is that `locales` is not getting added to an application bundle, i.e. there is no such directory in the unpacked app in `C:\Temp\_MEI85922`? Is there anything from PyInstaller log indicating this dir is added/not-added? Could it be PyInstaller issue?

**Comment by FeodorFitsner** [comment]
The code for `flet pack` is quite trivial, you can try looking at it to see if there is an issue: https://github.com/flet-dev/flet/blob/main/sdk/python/packages/flet/src/flet/cli/commands/pack.py#L183-L186

**Comment by pwd491** [comment]
> i.e. there is no such directory in the unpacked app in `C:\Temp\_MEI85922`

No, the directory exists and it can be seen from the logs, but the gettext module does not find the locale, I assume that the code is in main.py runs faster than unload all the source files to the tmp folder, although it does not sound logical

**Comment by FeodorFitsner** [comment]
I can't comment on the process of unpacking in PyInstaller.

To further investigate the issue to understand whether it comes from Flet, PyInstaller or gettext I would create a simple Python app, without Flet and with gettext, pack it with PyInstaller and debug.

**Comment by FeodorFitsner** [comment]
Closing the issue because there was no response from the author.
